src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java

```diff
@@ -97,6 +92,29 @@ public void testS3ClientCreateDestroy() {
         }
     }
 
+    @Test
```

Comment:
Can we add some tests that verify that options configurations actually behave as expected?

Reply:
Added some verification that the options are being passed correctly into the retry strategy.  I think validating further behavior of the retry strategy is probably out of scope in the CRT, but open to chat about it.